### PR TITLE
add dryrun param for adding repo

### LINKED ISSUE #99: Support dryrun option on repo add
anchore-engine v0.8.0 API supports listing tags in a repo without adding them to the system as a way to display a preview of the tags that would be added and could be used to warn users from accidentally adding large repos. Add a new option to `anchore-cli repo add` to expose this functionality

### PR DESCRIPTION
**Pull request by dakaneye**
Fixes #99 
Signed-off-by: Samuel Dacanay <sam.dacanay@anchore.com>

### CODE REVIEW

File: anchorecli/cli/repo.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import sys
 import json
+import sys
```
Comment (alfredodeza) [inline anchorecli/cli/repo.py:2]: why is this changing? from the linter?

File: anchorecli/clients/apiexternal.py [hunk 2]
```diff
@@ -830,31 +831,34 @@ def get_subscription_types(config):
...
     base_url = config['url']
 
-    ret = {}
+    url = os.path.join(base_url, 'repositories')
 
-    base_url = re.sub("/$", "", base_url)
-    url = '/'.join([base_url, "repositories?repository="+input_repo+"&autosubscribe="+str(autosubscribe)])
-    if lookuptag:
-        url = url + "&lookuptag="+str(lookuptag)
+    query = {
```
Comment (alfredodeza) [inline anchorecli/clients/apiexternal.py:842]: perfect

File: anchorecli/clients/apiexternal.py [hunk 3]
```diff
@@ -830,31 +831,34 @@ def get_subscription_types(config):
...
-        r = requests.post(url, auth=(userId, password), verify=config['ssl_verify'], headers=header_overrides)
+        r = requests.post(
+            url, auth=(user_id, password), verify=config['ssl_verify'], headers=header_overrides, params=query
+        )
         ret = anchorecli.clients.common.make_client_result(r, raw=False)
     except Exception as err:
         raise err
 
-    return(ret)
-    #return(add_subscription(config, 'repo_update', input_repo))
```
Comment (alfredodeza) [inline anchorecli/clients/apiexternal.py:857]: thanks for removing this

File: anchorecli/clients/apiexternal.py [hunk 4]
```diff
@@ -830,31 +831,34 @@ def get_subscription_types(config):
...
 
     try:
         _logger.debug("POST url=%s", str(url))
-        r = requests.post(url, auth=(userId, password), verify=config['ssl_verify'], headers=header_overrides)
+        r = requests.post(
+            url, auth=(user_id, password), verify=config['ssl_verify'], headers=header_overrides, params=query
+        )
         ret = anchorecli.clients.common.make_client_result(r, raw=False)
     except Exception as err:
         raise err
```
Comment (alfredodeza) [inline anchorecli/clients/apiexternal.py:859]: this can be removed though, doesn't do anything

**Review by alfredodeza — CHANGES_REQUESTED** [review]

File: anchorecli/cli/repo.py (hunk 1 above)
Comment (dakaneye) [reply anchorecli/cli/repo.py:2]: I think so

File: anchorecli/clients/apiexternal.py (hunk 4 above)
Comment (dakaneye) [reply anchorecli/clients/apiexternal.py]: fixed!

File: anchorecli/clients/apiexternal.py (hunk 2 above)
Comment (zhill) [reply anchorecli/clients/apiexternal.py:842]: Url join is available that will be more correct than the os path. Not sure if it's available for 2.7. https://docs.python.org/3/library/urllib.parse.html

**Pull request merged**